client/interfaces.go

```diff
@@ -64,9 +64,15 @@ type Interfaces struct {
 	Slots []Slot `json:"slots"`
 }
 
-// InterfaceMetaData contains meta-data about a given interface type.
-type InterfaceMetaData struct {
+// Interface holds information about a given interface and its instances.
+type Interface struct {
+	Name        string `json:"name,omitempty"`
+	Summary     string `json:"summary,omitempty"`
 	Description string `json:"description,omitempty"`
+	DocsURL     string `json:"docs-url,omitempty"`
+	Plugs       []Plug `json:"plugs,omitempty"`
+	Slots       []Slot `json:"slots,omitempty"`
+	Used        bool   `json:"used,omitempty"`
```

Comment:
This is so that we can ask for `snap interface --all` and see anything our snapd understands, even if we don't use it.